packages/discovery/src/discovery/provider/ProviderWithCache.test.ts

```diff
@@ -0,0 +1,169 @@
+import { expect, mockFn, mockObject } from 'earl'
+import { providers } from 'ethers'
+
+import { ChainId } from '../../utils/ChainId'
+import { EtherscanLikeClient } from '../../utils/EtherscanLikeClient'
+import { DiscoveryLogger } from '../DiscoveryLogger'
+import { DiscoveryCache, ProviderWithCache } from './ProviderWithCache'
+
+function setupProviderWithMockCache(values: {
+  curBlockNumber: number
+  reorgSafeDepth: number | undefined
+}) {
+  const mockCache = mockObject<DiscoveryCache>({
+    set: mockFn().resolvesTo(undefined),
+    get: mockFn()
+      .given('mockCachedKey')
+      .resolvesToOnce('mockCachedValue')
+      .given('mockNotCachedKey')
+      .resolvesToOnce(undefined),
+  })
+  const mockProvider = mockObject<providers.Provider>({
+    getBlockNumber: mockFn().resolvesTo(values.curBlockNumber),
+  })
+  const providerWithCache = new ProviderWithCache(
+    mockProvider,
+    mockObject<EtherscanLikeClient>({}),
+    DiscoveryLogger.SILENT,
+    ChainId.ETHEREUM,
+    mockCache,
+    undefined,
+    values.reorgSafeDepth,
+  )
+  return { providerWithCache, mockCache }
+}
+
+describe('ProviderWithCache', () => {
```

Comment:
```suggestion
describe(ProviderWithCache.name, () => {
```

More refactor friendly